File: csharp/HttpUtils.cs
using System.Net;
using System.Text;
using System.Threading.Tasks;

namespace languages
{
    public static class HttpUtils
    {
        internal const string mainPage = "/index.html";

        internal const string ResponseTemplate = "<HTML><BODY>{0}</BODY></HTML>";

        internal const string ResponseWithScriptTemplate = "<HTML>" + ScriptTemplate + "<BODY>{1}</BODY></HTML>";
        internal const string ScriptTemplate = "<script>{0}</script>";

        internal const string redirectTimerScript = "setTimeout(function () {window.location.href = '" + mainPage + "';}, 3000);";

        internal static async Task WriteResponseTextAsync(this HttpListenerResponse response, string text)
        {
            string responsePayload = string.Format(HttpUtils.ResponseTemplate, text);
            var responseBytes = Encoding.UTF8.GetBytes(responsePayload);
            await response.OutputStream.WriteAsync(responseBytes);
        }

        internal static async Task JsRedirectWithMessageAsync(this HttpListenerResponse response, string text)
        {
            string responsePayload = string.Format(HttpUtils.ResponseWithScriptTemplate,
                HttpUtils.redirectTimerScript,
                text);

            var responseBytes = Encoding.UTF8.GetBytes(responsePayload);
            await response.OutputStream.WriteAsync(responseBytes);
        }
    }
}

File: kotlin/src/main/kotlin/HttpUtils.kt
import com.sun.net.httpserver.HttpExchange
import java.net.URI

class HttpUtils {
    companion object {
        private const val mainPage: String = "/index.html"

        private const val responseTemplate: String = "<HTML><BODY>%s</BODY></HTML>"

        private const val scriptTemplate: String = "<script>%s</script>"
        private const val responseWithScriptTemplate: String = "<HTML>$scriptTemplate<BODY>%s</BODY></HTML>"

        private const val redirectTimerScript: String =
            "setTimeout(function () {window.location.href = '$mainPage';}, 3000);"

        fun HttpExchange.writeResponseText(text: String) {
            val bytesToSend = responseTemplate.format(text).toByteArray()
            this.sendResponseHeaders(200, 0)
            this.responseBody.use {
                it.write(bytesToSend)
            }
        }

        fun HttpExchange.jsRedirectWithMessage(message: String) {
            val bytesToSend = responseWithScriptTemplate.format(redirectTimerScript, message)
                .toByteArray()
            this.sendResponseHeaders(200, 0)
            this.responseBody.use { it.write(bytesToSend) }
        }

        fun URI.getQueryStringParameters(): Map<String, String> {
            return this.query?.split("&")
                ?.map { it.split("=") }
                ?.associate { it.first() to it.last() } ?: emptyMap()
        }
    }
}